src/gui/tray/activitylistmodel.h

```diff
@@ -117,55 +108,77 @@ public slots:
     void slotTriggerAction(const int activityIndex, const int actionIndex);
     void slotTriggerDismiss(const int activityIndex);
 
+    void addNotificationToActivityList(Activity activity);
+    void addErrorToActivityList(Activity activity);
+    void addIgnoredFileToList(Activity newActivity);
+    void addSyncFileItemToActivityList(Activity activity);
+    void removeActivityFromActivityList(int row);
+    void removeActivityFromActivityList(Activity activity);
+
+    void setAccountState(AccountState *state);
+    void setReplyMessageSent(const int activityIndex, const QString &message);
+    void setCurrentItem(const int currentItem);
+
 signals:
     void activityJobStatusCode(int statusCode);
     void sendNotificationRequest(const QString &accountName, const QString &link, const QByteArray &verb, int row);
 
 protected:
-    void setup();
-    void activitiesReceived(const QJsonDocument &json, int statusCode);
     QHash<int, QByteArray> roleNames() const override;
 
-    void setAndRefreshCurrentlyFetching(bool value);
     bool currentlyFetching() const;
+
+    const ActivityList &finalList() const; // added for unit tests
+
+protected slots:
+    void activitiesReceived(const QJsonDocument &json, int statusCode);
+    void setAndRefreshCurrentlyFetching(bool value);
     void setDoneFetching(bool value);
     void setHideOldActivities(bool value);
     void setDisplayActions(bool value);
+    void setFinalList(const ActivityList &finalList); // added for unit tests
 
     virtual void startFetchJob();
 
-    // added these for unit tests
-    void setFinalList(const ActivityList &finalList);
-    const ActivityList &finalList() const;
-    int currentItem() const;
-    //
-
 private:
+    enum class ActivityEntryType {
```

Comment:
should still get the `Q_ENUMS` macro in order to get improved logging feature added by it

Reply:
I added the macro